include/seqan3/std/iterator

```diff
@@ -19,13 +19,17 @@
 #ifdef __cpp_lib_ranges
 namespace std
 {
-inline namespace cpp20
+namespace cpp20
```

Comment:
removing inline was necessary, because?

Reply:
I normally write my reasons in the commit itself:

https://github.com/seqan/seqan3/pull/1909/commits/37dfe17eeda35476d2afb16c7824cb8e098c061f

> It turned out that inline namespace for `std::cpp20` was a bad idea
since it produced ambiguousness for `std::ostream_iterator` and
`std::cpp20::ostream_iterator`.

Reply:
> I normally write my reasons in the commit itself:

I am usually not going through the commits.